test/e2e/framework/service/jig.go

```diff
@@ -82,6 +111,61 @@ func NewTestJig(client clientset.Interface, namespace, name string) *TestJig {
 	return j
 }
 
+// allocatePort reserves the port provided as input.
+// If an invalid port was provided or if the port is already reserved, it returns false
+func (s *staticPortRange) allocatePort(port int) bool {
+	s.Lock()
+	defer s.Unlock()
+	port -= s.baseport
+	if port < 0 || port > s.length || s.reservedPorts.Has(port) {
+		return false
+	}
+	s.reservedPorts.Insert(port)
+	return true
+}
+
+// nextFreePort returns a free port from the range and returns its number and true
+// the port is not allocated so the consumer should allocate it explicitly calling allocatePort()
+// if none is available then it returns -1 and false
+func (s *staticPortRange) nextFreePort() (int, bool) {
+	s.Lock()
+	defer s.Unlock()
+	// start in a random offset
+	start := rand.Intn(s.length)
+	for i := 0; i < s.length; i++ {
+		port := (start + i) % s.length
+		if !s.reservedPorts.Has(port) {
+			return s.baseport + port, true
+		}
+	}
+	return -1, false
+}
+
+// release the port passed as an argument
+func (s *staticPortRange) release(port int) {
+	s.Lock()
+	defer s.Unlock()
+	port -= s.baseport
+	s.reservedPorts.Delete(port)
+}
+
+// ReserveStaticNodePort reserves the port provided as input.
+// If an invalid port was provided or if the port is already reserved, it returns false
+func ReserveStaticNodePort(port int) bool {
+	return staticPortAllocator.allocatePort(port)
+}
+
+// GetUnusedStaticNodePort returns a free port in static range and a boolean True value
+// If no port in static range is available it returns -1 and a boolean value False
```

Comment:
```suggestion
// GetUnusedStaticNodePort returns a free port in static range and a boolean True value.
// If no port in static range is available it returns -1 and a boolean value False.
// Note that it is not guaranteed that the returned port is actually available on the apiserver;
// You must allocate a port, then attempt to create the service, then call
// ReserveStaticNodePort.
```

(and maybe move this before ReserveStaticNodePort, so the functions appear in the order you use them?)